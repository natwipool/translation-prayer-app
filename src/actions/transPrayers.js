// SET_TRANS_PRAYER
const setTransPrayer = (transPrayers = {}) => ({
  type: 'SET_TRANS_PRAYER',
  transPrayers
});

export default setTransPrayer;