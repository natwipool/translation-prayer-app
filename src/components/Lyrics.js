import React from 'react';
import Lyric from './Lyric';

const Lyrics = (props) => (
  <div>
    {
      props.transPrayers.length === 0 ? (
        <p>ไม่มีบทสวด</p>
      ) : (
        props.transPrayers.filter((transPrayer, index) => 
          index  === props.index
        ).map((result, index) => 
          <Lyric
            key={index} 
            transPrayer={result}
          />
        )
      )
    }
  </div>
)

export default Lyrics;