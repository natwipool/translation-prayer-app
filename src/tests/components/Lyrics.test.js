import React from 'react';
import { shallow } from 'enzyme';
import Lyrics from '../../components/Lyrics';
import transPrayers from '../fixtures/transPrayers';

test('should render Lyrics correctly', () => {
  const wrapper = shallow(<Lyrics transPrayers={transPrayers} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Lyrics with no playlist', () => {
  const wrapper = shallow(<Lyrics transPrayers={[]} />);
  expect(wrapper).toMatchSnapshot();
});