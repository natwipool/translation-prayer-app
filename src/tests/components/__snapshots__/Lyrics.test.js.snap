// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Lyrics correctly 1`] = `<div />`;

exports[`should render Lyrics with no playlist 1`] = `
<div>
  <p>
    ไม่มีบทสวด
  </p>
</div>
`;
